src/lib/operators/aggregate_hash.cpp

```diff
@@ -832,8 +832,34 @@ std::shared_ptr<const Table> AggregateHash::_on_execute() {
     const auto data_type =
         input_column_id == INVALID_COLUMN_ID ? DataType::Long : input_table->column_data_type(input_column_id);
 
-    resolve_data_type(data_type, [&, aggregate_idx](auto type) {
-      _write_aggregate_output(type, aggregate_idx, aggregate->aggregate_function);
+    resolve_data_type(data_type, [&](auto type) {
+      using ColumnDataType = typename decltype(type)::type;
```

Comment:
Did clang-tidy complain or why was the code moved?

Reply:
clang-tidy complained about the unused `type` argument for `_write_aggregate_output`. After removing it, I did not see a reason to call `_write_aggregate_output<DataType>(column, function)` here which only contained of a single switch to forward it to `_write_aggregate_output<DataType, Function>(column)` (so got rid one one method call). Though I don't think there will be any performance impact, I'm gonna run the benchmarks. Besides, I noticed we could make the latter protected.

Reply:
(same for `JoinHash::calculate_radix_bits(...)`: could be made protected if the test was a friend).

Reply:
Hmm, I don't see value in having protected methods. But that's a personal thing.

Reply:
People of at least two chairs (SWA, SAM) are starting heavy breathing/have a heart attack right now.